Source/Core/loadKTX2.js

```diff
@@ -0,0 +1,114 @@
+ * <p>
+ * The following are part of the KTX2 format specification but are not supported:
+ * <ul>
+ *     <li>Metadata</li>
+ *     <li>3D textures</li>
+ *     <li>Texture Arrays</li>
+ *     <li>Video</li>
+ *     <li>Compressed Mipmaps</li>
+ * </ul>
+ * </p>
+ *
+ * @exports loadKTX2
+ *
+ * @param {Resource|String|ArrayBuffer} resourceOrUrlOrBuffer The URL of the binary data or a ArrayBuffer.
+ * @returns {Promise.<CompressedTextureBuffer>|undefined} A promise that will resolve to the requested data when loaded. Returns undefined if <code>request.throttle</code> is true and the request does not have high enough priority.
+ *
+ * @exception {RuntimeError} Invalid KTX2 file.
+ * @exception {RuntimeError} File is the wrong endianness.
+ * @exception {RuntimeError} glInternalFormat is not a valid format.
+ * @exception {RuntimeError} glType must be zero when the texture is compressed.
+ * @exception {RuntimeError} The type size for compressed textures must be 1.
+ * @exception {RuntimeError} glFormat must be zero when the texture is compressed.
+ * @exception {RuntimeError} Generating mipmaps for a compressed texture is unsupported.
+ * @exception {RuntimeError} The base internal format must be the same as the format for uncompressed textures.
+ * @exception {RuntimeError} 3D textures are not supported.
+ * @exception {RuntimeError} Texture arrays are not supported.
+ * @exception {RuntimeError} Cubemaps are not supported.
+ *
+ * @example
+ * // load a single URL asynchronously
+ * Cesium.loadKTX2('some/url').then(function (ktx2Data) {
+ *     var width = ktx2Data.width;
+ *     var height = ktx2Data.height;
+ *     var format = ktx2Data.internalFormat;
+ *     var arrayBufferView = ktx2Data.bufferView;
+ *     // use the data to create a texture
+ * }).otherwise(function (error) {
+ *     // an error occurred.
+ * });
+ *
+ * @see {@link https://github.com/KhronosGroup/KTX-Specification|KTX file format}
+ * @see {@link http://www.w3.org/TR/cors/|Cross-Origin Resource Sharing}
+ * @see {@link http://wiki.commonjs.org/wiki/Promises/A|CommonJS Promises/A}
+ * @private
+ */
+function loadKTX2(resourceOrUrlOrBuffer) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.defined("resourceOrUrlOrBuffer", resourceOrUrlOrBuffer);
+  //>>includeEnd('debug');
+
+  var loadPromise;
+  if (
+    resourceOrUrlOrBuffer instanceof ArrayBuffer ||
+    ArrayBuffer.isView(resourceOrUrlOrBuffer)
+  ) {
+    loadPromise = when.resolve(resourceOrUrlOrBuffer);
+  } else {
+    var resource = Resource.createIfNeeded(resourceOrUrlOrBuffer);
+    loadPromise = resource.fetchArrayBuffer();
+  }
+
+  if (!defined(loadPromise)) {
+    return undefined;
+  }
+
+  // load module then return
+  return loadPromise.then(function (data) {
+    if (defined(data)) {
+      // data array can be a view of a shared buffer,
+      // so make a copy that can be given to the worker.
+      var copy = arraySlice(data);
```

Comment:
I think it's safe to remove `arraySlice` here because the ktx2 buffers are always coming from files that are loaded just for `loadKTX2` - testing without arraySlice here works fine in Sandcastle and specs (which all call loadKTX2 with a resource or url). Would there ever be a case where the data passed to `loadKTX2` would be needed somewhere else too?

Reply:
A KTX2 could be embedded in a glb. Does that still work? I wouldn't expect it to be a problem.

Reply:
Just tested this with a .glb and it works fine. I added the .glb to the ktx2 sample data folder just in case.